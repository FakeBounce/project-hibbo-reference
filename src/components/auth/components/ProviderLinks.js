import React from 'react';
import { Image, Text } from 'react-native';
import {
  bool,
  func,
  string,
  number,
  oneOfType,
  object,
  array,
} from 'prop-types';

import hourglass from 'assets/shared/json/hourglass.json';

import { SRow, SButtonProvider, SProviderAnim } from '../styles';

const ProviderLinks = ({
  name,
  bordered,
  onPress,
  handlePosition,
  elementPosition,
  imageSource,
  imageStyle,
  label,
  labelStyle,
  isButtonDisabled,
  authProvider,
}) => {
  return (
    <SButtonProvider
      bordered={bordered}
      activeOpacity={1}
      onLayout={({ nativeEvent: { layout: { x, y } } }) => {
        handlePosition(elementPosition, { x, y });
      }}
      onPress={() => {
        onPress();
      }}
      disabled={isButtonDisabled}
    >
      {name === authProvider ? (
        <SRow>
          <SProviderAnim
            innerRef={anim => {
              if (anim) {
                anim.play();
              }
            }}
            source={hourglass}
            loop
          />
        </SRow>
      ) : (
        <SRow>
          <Image style={imageStyle} source={imageSource} />
          <Text style={labelStyle}>{label}</Text>
        </SRow>
      )}
    </SButtonProvider>
  );
};

ProviderLinks.defaultProps = {
  bordered: false,
  elementPosition: '',
  handlePosition: () => {},
};

ProviderLinks.propTypes = {
  name: string.isRequired,
  bordered: bool,
  onPress: func.isRequired,
  handlePosition: func,
  imageSource: oneOfType([object, number, array]).isRequired,
  label: string.isRequired,
  elementPosition: string,
  labelStyle: oneOfType([object, number, array]).isRequired,
  imageStyle: oneOfType([object, number, array]).isRequired,
  isButtonDisabled: bool.isRequired,
  authProvider: string.isRequired,
};

export default ProviderLinks;
