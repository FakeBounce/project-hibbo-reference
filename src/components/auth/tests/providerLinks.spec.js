import React from 'react';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import source from 'assets/auth/icons/mail.png';

import ProviderLinks from '../components/ProviderLinks';

describe('ProviderLinks', () => {
  const fn = jest.fn();

  it('should render the component', () => {
    const component = renderer
      .create(
        <ProviderLinks
          bordered
          onPress={fn}
          handlePosition={() => {}}
          elementPosition=""
          imageSource={source}
          imageStyle={0}
          label="Label"
          labelStyle={0}
        />,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should handle press events', () => {
    const tree = shallow(
      <ProviderLinks
        bordered
        onPress={fn}
        handlePosition={() => {}}
        elementPosition=""
        imageSource={source}
        imageStyle={0}
        label="Label"
        labelStyle={0}
      />,
    );

    tree.find(TouchableOpacity).at(0).simulate('press');
    expect(fn).toHaveBeenCalled();
  });

  it('should trigger onLayout fun', () => {
    shallow(
      <ProviderLinks
        bordered
        handlePosition={fn}
        onPress={() => {}}
        elementPosition="fbPosition"
        imageSource={source}
        imageStyle={0}
        label="Label"
        labelStyle={0}
      />,
    );

    expect(fn).toHaveBeenCalled();
  });
});
