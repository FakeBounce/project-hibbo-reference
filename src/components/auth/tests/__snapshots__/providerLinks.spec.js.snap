// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderLinks should render the component 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={1}
    style={0}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={0}
  >
    Label
  </Text>
</View>
`;
